url-updater/src/main/java/com/devonfw/tools/ide/url/updater/python/PythonUrlUpdater.java

```diff
@@ -1,38 +1,108 @@
+import com.devonfw.tools.ide.url.model.folder.UrlTool;
 import com.devonfw.tools.ide.url.model.folder.UrlVersion;
-import com.devonfw.tools.ide.url.updater.WebsiteUrlUpdater;
+import com.devonfw.tools.ide.url.updater.JsonUrlUpdater;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+
+public class PythonUrlUpdater extends JsonUrlUpdater<PythonJsonObject> {
+
+    /**
+     * The bas Url of the version json
+     */
+    private String VERSION_BASE_URL = "https://raw.githubusercontent.com";
+
+
+    private final static String VERSION_FILENAME ="actions/python-versions/main/versions-manifest.json";
+    final  static ObjectMapper MAPPER = JsonMapping.create();
+    private static final Logger logger = LoggerFactory.getLogger(PythonUrlUpdater.class);
+
+
+    @Override
+    protected String getTool() {
+        return "python";
+    }
+
+    @Override
+    protected void addVersion(UrlVersion urlVersion) {
+        throw new IllegalStateException();
+    }
+
+    protected String getVersionBaseUrl() {
+
+        return this.VERSION_BASE_URL;
+    }
+
+    @Override
+    protected String doGetVersionUrl() {
+        return getVersionBaseUrl() + "/"+ VERSION_FILENAME;
+    }
+
+    @Override
+    protected Class<PythonJsonObject> getJsonObjectType() {
+        return null;
+    }
+
+    @Override
+    protected void collectVersionsFromJson(PythonJsonObject jsonItem, Collection<String> versions) {
+
+    }
+
+
+    @Override
+    public void update(UrlRepository urlRepository) {
+        UrlTool tool = urlRepository.getOrCreateChild(getTool());
+
+
+        String url = doGetVersionUrl();
+        try {UrlEdition edition = tool.getOrCreateChild(getEdition());
+            updateExistingVersions(edition);
+            String toolWithEdition = getToolWithEdition();
+            String response = doGetResponseBodyAsString(url);
+             PythonJsonItem[] res = MAPPER.readValue(response, PythonJsonItem[].class);
+
+                 for (PythonJsonItem result : res) {
+                     String version = result.getVersion();
+                     System.out.println("version");
+                     if (edition.getChild(version) == null) {
+                         try {
+                             UrlVersion urlVersion = edition.getOrCreateChild(version);
+                             for (PythonFiles download : result.getFiles()) {
+                                 if (download.getPlatform().equals("win32")) {
+                                     doAddVersion(urlVersion, download.getDownload_url());
+                                 } else if (download.getPlatform().equals("linux")) {
+                                     doAddVersion(urlVersion, download.getDownload_url());
+                                 } else if (download.getPlatform().equals("darwin")) {
+                                     doAddVersion(urlVersion, download.getDownload_url());
```

Comment:
Shouldn't we add `os` and `architecture` here too?

Reply:
In the Json File every os has it's own DownloadUrl